lib/generateTangentsBinormals.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+var Cesium = require('cesium');
+var gltfPrimitiveToCesiumGeometry = require('./gltfPrimitiveToCesiumGeometry');
+var cesiumGeometryToGltfPrimitive = require('./cesiumGeometryToGltfPrimitive');
+
+var defined = Cesium.defined;
+var WebGLConstants = Cesium.WebGLConstants;
+var GeometryPipeline = Cesium.GeometryPipeline;
+
+module.exports = generateTangentsBinormals;
+
+/**
+ * Generates tangents and binormals for primitives if they do not exist.
+ *
+ * The glTF asset must be initialized for the pipeline. It mus already have normals.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset.
+ * @returns {Object} The glTF asset with generated tangents and binormals.
+ *
+ * @see addPipelineExtras
+ * @see loadGltfUris
+ */
+function generateTangentsBinormals(gltf) {    
+    var primitives = getPrimitives(gltf);
+    var primitivesLength = primitives.length;
+    for (var i = 0; i < primitivesLength; i++) {
+        var primitive = primitives[i];
+        var positionAccessorId = primitive.attributes.POSITION;
+        var normalAccessorId   = primitive.attributes.NORMAL;
+        var texcoordAccessorId = primitive.attributes.TEXCOORD_0;
+        var binormalAccessorId = primitive.attributes.BINORMAL;
+        var tangentAccessorId  = primitive.attributes.TANGENT;
+        var indicesAccessorId  = primitive.indices;
+
+        if (primitive.mode === WebGLConstants.TRIANGLES &&  defined(indicesAccessorId)  &&
+        												    defined(positionAccessorId) &&
+        												    defined(normalAccessorId)   &&
+        												    defined(texcoordAccessorId) &&
+        												   !defined(binormalAccessorId) &&
+        												   !defined(tangentAccessorId)    ) {
+            var geometry = gltfPrimitiveToCesiumGeometry(gltf, primitive);
+            GeometryPipeline.computeBinormalAndTangent(geometry);
+            cesiumGeometryToGltfPrimitive(gltf, primitive, geometry);
+        }
+    }
+    return gltf;
+}
+
+//TODO: copied from generateNormals, could be shared
```

Comment:
After a second look, I just saw that there's also a `getPrimitives` function in `PrimitiveHelpers.js`. I used that one now in `getNormals`and `getTangentsBinormals`:
https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/203/commits/ffa215c4f726f8a02edea5275025179fc537335e

The difference is, however, that this function accesses the primitives directly from the meshes, not via the nodes. I believe this makes more sense in this case - otherwise, when going over the nodes, we would potentially iterate over a single mesh multiple times, if it is being shared by several nodes?

I also moved the old function from `generateNormals`, which iterates over the nodes, into the `PrimitiveHelpers` file: https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/203/commits/ef141f0a4063d402021819ef4f3cfd93eb812484
However, right now, this function is not being used anywhere.

Does that make sense, should it be kept like this?